src/commands/org/list/shape.ts

```diff
@@ -44,26 +44,51 @@ export class OrgShapeListCommand extends SfCommand<OrgShapeListResult[]> {
   // there were no flags being used in the original!
   // eslint-disable-next-line sf-plugin/should-parse-flags
   public async run(): Promise<OrgShapeListResult[]> {
-    const shapes = await getAllOrgShapesFromAuthenticatedOrgs();
-    if (shapes.length === 0) {
+    const { orgShapes, errors } = await getAllOrgShapesFromAuthenticatedOrgs();
+    errors.forEach((e) => this.warn(e));
+    if (orgShapes.length === 0) {
+      this.log();
       this.info(messages.getMessage('noOrgShapes'));
-      return shapes;
+      return orgShapes;
     }
 
     this.styledHeader('Org Shapes');
     this.table(
-      shapes.map((shape) => (shape.status === 'Active' ? { ...shape, status: chalk.green(shape.status) } : shape)),
+      orgShapes.map((shape) => (shape.status === 'Active' ? { ...shape, status: chalk.green(shape.status) } : shape)),
       orgShapeColumns
     );
-    return shapes;
+    return orgShapes;
   }
 }
 
-export const getAllOrgShapesFromAuthenticatedOrgs = async (): Promise<OrgShapeListResult[]> => {
+export const getAllOrgShapesFromAuthenticatedOrgs = async (): Promise<{
+  orgShapes: OrgShapeListResult[];
+  errors: Error[];
+}> => {
   const orgs = await AuthInfo.listAllAuthorizations((orgAuth) => !orgAuth.error && !orgAuth.isScratchOrg);
   if (orgs.length === 0) {
     throw messages.createError('noAuthFound');
   }
-  const shapes = await Promise.all(orgs.map((o) => getAllShapesFromOrg(o)));
-  return shapes.flat();
+  const shapes = await Promise.allSettled(orgs.map((o) => getAllShapesFromOrg(o)));
+  // get all fulfilled promises values from settled promises
+  const orgShapes = shapes
+    .filter((s) => s.status === 'fulfilled')
```

Comment:
sucks that TS doesn't understand these quite yet.  Suggestion: make a typeguard that determines if it's an Error or an OrgShape based on the fulfilled property.

Then you can use it in both arrays and not have to check status in both the filter and map